src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
             },
             "file_name_col": {
                 "type": "string",
-                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date."
+                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date.",
             },
             "map": {
                 "type": "list",
                 "doc": "List of lists of unique publications references and publication names in the collection provided in the same order.",
                 "optional": True,
-            }
+            },
         }
 
     def create_outputs_schema(self):
         return {
             "corpus_table": {
                 "type": "table",
-                "doc": "Augmented table containing extracted metadata."
+                "doc": "Augmented table containing extracted metadata.",
             }
         }
 
     def process(self, inputs, outputs) -> None:
-        
+        import re
+        import polars as pl
+        import pyarrow as pa
+
         table_obj = inputs.get_value_obj("corpus_table")
         column_name = inputs.get_value_obj("file_name_col").data
-        pub_refs, pub_names = None, None
 
-        try:
-            pub_refs = inputs.get_value_obj("map").data[0]
-            pub_names = inputs.get_value_obj("map").data[1]
+        if not column_name:
+            raise KiaraProcessingException("No file name column name provided.")
+        
+        if not table_obj:
+            raise KiaraProcessingException("No table provided.")
+        
+        if table_obj.is_set:
 
-        except Exception as e:
-            pass;
+            sources: Union[pa.Table, None] = table_obj.data
+            
+            assert sources is not None
 
+            sources_col_names = sources.column_names
+
+            if column_name not in sources_col_names:
+
+                raise KiaraProcessingException(
+                    f"Could not find file names column '{column_name}' in the table. Please specify a valid column name manually, using one of: {', '.join(sources_col_names)}"
+                )
+            
+            sources_data: pa.Table = table_obj.data.arrow_table
```

Comment:
Just for clarity, here we use the `arrow_table` method from `KiaraTable` to get an Arrow table type. You could have just used:

```
sources_data = sources.arrow_table
```
Probably, in the interest of readability and explicity, the variable would be better named `sources_arrow_format` or something.